components/rendering/src/markdown.rs

```diff
@@ -25,6 +28,19 @@ pub struct Rendered {
     pub toc: Vec<Header>,
 }
 
+#[derive(Debug)]
+struct HeaderIndex {
+    start: usize,
+    end: usize,
+    level: i32,
+}
```

Comment:
can you add some comments on what the struct is and what the fields are? `level` is seemingly obvious  but even then since it's called `HeaderIndex` and not `Header`, it isn't obvious

Reply:
I renamed `HeaderIndex` to `HeaderRef`. is that better?

Reply:
```rust
// tracks a header in a slice of pulldown-cmark events
#[derive(Debug)]
struct HeaderRef {
    start_idx: usize,
    end_idx: usize,
    level: i32,
}
```